go/runtime/transaction/transaction_test.go

```diff
@@ -181,3 +181,111 @@ func TestTransactionInvalidBatchOrder(t *testing.T) {
 	_, err = tree.GetInputBatch(ctx, 0, 0)
 	require.Error(t, err, "GetInputBatch should fail with inconsistent order")
 }
+
+func TestIOWriteLogValidation(t *testing.T) {
```

Comment:
Could you change this function to do a table-based test instead? I think this would be more clear.